### DISCUSSION TITLE
How to prevent resources from direct access by typing URL

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6275] [Help] [answered]
Hi All,

Base on filament document 

[Disabling resource or page navigation items](https://filamentphp.com/docs/2.x/admin/navigation#disabling-resource-or-page-navigation-items)

To prevent resources or pages from showing up in navigation, you may use:

`protected static bool $shouldRegisterNavigation = false;`

-------------

This method only hide the resource on navigation,. but not protect for someone by typing directly on address bar.

How I can do it real protection for my resource ?

Thank you

**Comment by VincvanAs** [comment]
public function mount(): void
{
    abort_unless(auth()->user()->canManageSettings(), 403);
}

**Reply by danharrin** [reply, marked as answer]
This works for custom pages. For resources, you just need to use a policy on the model to control the `viewAny()` permission, which also handles navigation.
